app/main/helpers/search_helpers.py

```diff
@@ -4,8 +4,9 @@
 from werkzeug.datastructures import MultiDict
 
 
-def get_lot_from_request(request):
-    return request.args.get('lot', None)
+def get_lot_from_request(request, all_lots):
+    lot = request.args.get('lot', None)
+    return lot if lot in all_lots or not all_lots else None
```

Comment:
And I know Python hates brackets but some parentheses might help make it more readable?
`lot if (not all_lots or lot in all_lots) else None`